packages/wrangler/src/dev/start-server.ts

```diff
@@ -95,7 +95,7 @@ export async function startDevServer(
 				port: props.inspectorPort,
 			},
 			urlOverrides: {
-				secure: props.localProtocol === "https",
+				secure: props.upstreamProtocol === "https",
```

Comment:
Currently `urlOverrides` is not used in the codebase but is expected to be used once we land the next phase of the startDev work, so this fix is needed here too.